src/bundler.ts

```diff
@@ -300,37 +310,53 @@ export class Bundler {
       reachedImports.add(resolvedUrl);
       this.rewriteImportedUrls(importDoc, resolvedUrl, docUrl);
 
-      let importParent: ASTNode;
       // TODO(usergenic): remove the remove() call when PolymerLabs/dom5#35 is
       // fixed
-      if (matchers.afterHiddenDiv(htmlImport)) {
-        importParent = dom5.nodeWalkPrior(htmlImport, matchers.hiddenDiv)!;
-        dom5.remove(htmlImport);
-        dom5.append(importParent, htmlImport);
-      } else if (matchers.beforeHiddenDiv(htmlImport)) {
-        const index = htmlImport.parentNode!.childNodes!.indexOf(htmlImport);
-        importParent = htmlImport.parentNode!.childNodes![index + 1];
-        dom5.remove(htmlImport);
-        astUtils.prepend(importParent, htmlImport);
-      } else if (!matchers.inHiddenDiv(htmlImport)) {
-        const hiddenDiv = this.createHiddenContainerNode();
-        dom5.replace(htmlImport, hiddenDiv);
-        dom5.append(hiddenDiv, htmlImport);
-        importParent = hiddenDiv;
+      let hiddenDiv = dom5.query(body, matchers.hiddenDiv);
+      if (!hiddenDiv) {
+        hiddenDiv = this.createHiddenContainerNode();
+        dom5.append(body, hiddenDiv);
+      }
+
+      if (hide) {
+        if (matchers.afterHiddenDiv(htmlImport)) {
+          dom5.remove(htmlImport);
+          dom5.append(hiddenDiv, htmlImport);
+        } else if (matchers.beforeHiddenDiv(htmlImport)) {
+          dom5.remove(htmlImport);
+          astUtils.prepend(hiddenDiv, htmlImport);
+        } else if (!matchers.inHiddenDiv(htmlImport)) {
+          dom5.remove(htmlImport);
+          dom5.append(hiddenDiv, htmlImport);
+        }
+
       } else {
-        importParent = htmlImport.parentNode!;
+        if (matchers.inHiddenDiv(htmlImport)) {
+          dom5.remove(htmlImport);
+          astUtils.insertAfter(hiddenDiv, htmlImport);
+        }
       }
 
-      /**
-       * Find the list of imports, then move the imports to the parent doc, then
-       * inline them.
-       *
-       * Finding them first saves on deduplication, moving them before inlining
-       * them
-       * allows the hidden div to always be found.
-       */
       const nestedImports = dom5.queryAll(importDoc, matchers.htmlImport);
-      astUtils.insertAllBefore(importParent, htmlImport, importDoc.childNodes!);
+
+      // If we're not hiding the import, because we're building the bundle
```

Comment:
nit: these two comments were a little hard to follow/understand. Would love if they could be cleaned up a bit